backend/game/scribble/handler/draw/draw.go

```diff
@@ -0,0 +1,160 @@
+package draw
+
+import (
+	"encoding/json"
+	"sync"
+
+	"github.com/bseto/arcade/backend/game"
+	"github.com/bseto/arcade/backend/log"
+	"github.com/bseto/arcade/backend/websocket/identifier"
+	"github.com/bseto/arcade/backend/websocket/registry"
+)
+
+const (
+	name = "draw"
+)
+
+type ReceiveDraw struct {
+	Action         DrawAction `json:"action"`
+	RequestHistory bool       `json:"requestHistory"`
+}
+
+type DrawReply struct {
+	Action  DrawAction   `json:"action"`
+	History *DrawHistory `json:"history,omitempty"`
+}
+
+type DrawHistory struct {
+	History []DrawAction `json:"history,omitempty"`
+}
+
+type DrawAction struct {
+	From       Point      `json:"from"`
+	To         Point      `json:"to"`
+	BrushStyle BrushStyle `json:"brushStyle"`
+	LineCap    LineCap    `json:"lineCap"`
+}
+
+type Point struct {
+	X int32 `json:"x"`
+	Y int32 `json:"y"`
+}
+
+type BrushStyle struct {
+	BrushSize  int32  `json:"brushSize"`
+	BrushColor string `json:"brushColor"`
+}
+
+type LineCap string
+
+const (
+	Round LineCap = "round"
+)
+
+type Handler struct {
+	drawHistoryLock sync.RWMutex
+	drawHistory     DrawHistory
+}
+
+func Get() *Handler {
+	return &Handler{}
+}
+
+// HandleInteraction will be given the tools it needs to handle
+// any interaction
+func (h *Handler) HandleInteraction(
+	message json.RawMessage,
+	caller identifier.Client,
+	registry registry.Registry,
+) {
+	var msg ReceiveDraw
+	err := json.Unmarshal(message, &msg)
+	if err != nil {
+		log.Errorf("unable to unmarshal message: %v", err)
+		return
+	}
+
+	if msg.RequestHistory == true {
+		h.SendHistory(caller, registry)
+	} else {
+		h.ForwardAction(msg.Action, caller, registry)
+	}
+
+	return
+}
+
+func (h *Handler) forwardAction(
+	drawAction DrawAction,
+	clientID identifier.Client,
+	reg registry.Registry,
+) {
+
+	drawBytes, err := game.MessageBuild(name, drawAction)
+	if err != nil {
+		log.Errorf("unable to build message: %v", err)
+		return
+	}
+
+	reg.SendToSameHubExceptCaller(clientID, drawBytes)
+}
+
+func (h *Handler) NewClient(
+	clientID identifier.Client,
+	reg registry.Registry,
+) {
+	// we don't need to send history on a new connection
+	//h.SendHistory(clientID, reg)
+}
+
+func (h *Handler) ClientQuit(
+	clientID identifier.Client,
+	reg registry.Registry,
+) {
+	// stub
+}
+
+// Name needs to return a unique name of this GameHandler
+// This return will be used for routing
+func (h *Handler) Name() string {
+	return name
+}
+
+func (h *Handler) SendHistory(
+	clientID identifier.Client,
+	reg registry.Registry,
+) {
+	h.drawHistoryLock.RLock()
+	defer h.drawHistoryLock.RUnlock()
+
+	historyBytes, err := game.MessageBuild(name, h.drawHistory)
+	if err != nil {
+		log.Errorf("unable to build message: %v", err)
+		return
+	}
+
+	reg.SendToCaller(clientID, historyBytes)
+}
+
+func (h *Handler) ForwardAction(
+	drawAction DrawAction,
+	caller identifier.Client,
+	registry registry.Registry,
+) {
+	log.Infof("Forwarding draw action: ", drawAction)
```

Comment:
mmm I think if you want this logged, maybe log it as Debugf. I don't want to see this on the log for every user unless i'm in debug mode (although I don't have a debug flag yet in main xD)

Reply:
I'll just take them off, I was kinda unsure if I should keep this or not.